mlflow/utils/databricks_utils.py

```diff
@@ -245,11 +246,86 @@ def is_in_databricks_runtime():
     return get_databricks_runtime_version() is not None
 
 
-def is_in_databricks_serverless():
+def is_in_databricks_serverless_runtime():
     dbr_version = get_databricks_runtime_version()
     return dbr_version and dbr_version.startswith("client.")
 
 
+def is_in_databricks_shared_cluster_runtime():
+    from mlflow.utils.spark_utils import is_spark_connect_mode
+    return (
+        is_in_databricks_runtime()
+        and is_spark_connect_mode()
+        and not is_in_databricks_serverless_runtime()
+    )
+
+
+def is_databricks_connect(spark):
+    from mlflow.utils.spark_utils import is_spark_connect_mode
+    """
+    Return True if current Spark-connect client connects to Databricks cluster.
+    """
+    # TODO: Remove the `spark.client._builder._build` attribute access once
+    #  Spark-connect has public attribute for this information.
+    return is_spark_connect_mode() and (
+        is_in_databricks_serverless_runtime() or
+        is_in_databricks_shared_cluster_runtime() or
+        "databricks-session" in spark.client._builder.userAgent
+    )
+
+
+_dbconnect_client = None
+_dbconnect_udf_sandbox_image_version = None
+_dbconnect_udf_sandbox_platform_machine = None
+
+
+def get_dbconnect_udf_sandbox_image_version_and_platform_machine(spark):
```

Comment:
can we split this function?

Reply:
I consider to merge them into one function because of the caching logic (see `_cached_dbconnect_client`, `_dbconnect_udf_sandbox_image_version`, `_dbconnect_udf_sandbox_platform_machine`) splitting it causing it hard to handle the caching